### PR TITLE
hdparm: name -> pname

### PR DESCRIPTION
**Pull request by Luflosi**
###### Motivation for this change
According to the manual, this is preferred for packages in Nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by omasanori — APPROVED** [review]

**Comment by omasanori** [comment]
Result of `nixpkgs-review pr 101582` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>gparted</li>
    <li>hdparm</li>
    <li>tlp</li>
  </ul>
</details>

File: pkgs/os-specific/linux/hdparm/default.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
 { stdenv, fetchurl }:
 
 stdenv.mkDerivation rec {
-  name = "hdparm-9.58";
+  pname = "hdparm";
+  version = "9.58";
 
   src = fetchurl {
-    url = "mirror://sourceforge/hdparm/${name}.tar.gz";
+    url = "mirror://sourceforge/hdparm/hdparm-${version}.tar.gz";
```
Comment (omasanori) [inline pkgs/os-specific/linux/hdparm/default.nix:8]:
```suggestion
    url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.gz";
```

**Review by omasanori — COMMENTED** [review]
LGTM with a small nit.

File: pkgs/os-specific/linux/hdparm/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/os-specific/linux/hdparm/default.nix:8]: if this is to remain stable in the future, I think ti's fine as is

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 101582 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>gparted</li>    <li>hdparm</li>    <li>tlp</li>  </ul></details>

**Pull request merged**